src/main/java/mate/academy/store/service/impl/UserServiceImpl.java

```diff
@@ -0,0 +1,30 @@
+package mate.academy.store.service.impl;
+
+import lombok.RequiredArgsConstructor;
+import mate.academy.store.dto.UserRegistrationRequestDto;
+import mate.academy.store.dto.UserResponseDto;
+import mate.academy.store.exception.RegistrationException;
+import mate.academy.store.mapper.UserMapper;
+import mate.academy.store.model.User;
+import mate.academy.store.repository.user.UserRepository;
+import mate.academy.store.service.UserService;
+import org.springframework.stereotype.Service;
+
+@Service
+@RequiredArgsConstructor
+public class UserServiceImpl implements UserService {
+    private final UserRepository userRepository;
+    private final UserMapper userMapper;
+
+    @Override
```

Comment:
Should we use `@Transactional` here?

Reply:
> Should we use `@Transactional` here?

I think yes, because if a RegistrationException is thrown, it is needed to perform a rollback

Reply:
Transactional annotation doesn't provide auto rollback. If you need to rollback transaction you should use appropriate attribute of this annotation. But this is not such case. Transactional is needed there because you call DB and save user to DB and in the next HWs you add one more query to DB